mkosi/__init__.py

```diff
@@ -5588,24 +5658,86 @@ def make_build_dir(args: CommandLineArguments) -> None:
     mkdir_last(args.build_dir, 0o755)
 
 
+def setup_ssh(args: CommandLineArguments, root: str, do_run_build_script: bool, for_cache: bool) -> Optional[TextIO]:
+    if do_run_build_script or for_cache or not args.ssh:
+        return None
+
+    if args.distribution in [Distribution.ubuntu, Distribution.debian]:
+        unit = "ssh"
+    else:
+        unit = "sshd"
```

Comment:
I'm not sure, this is necessary, at least on the Debian systems I run `sshd.service` is a valid alias to `ssh.service`

Reply:
I'll try removing this again but I'm pretty sure I got crashes somewhere without this.

Reply:
sshd.service is an alias of ssh.service - there's also ssh.socket for socket-based activation

```
$ systemctl cat ssh.service
# /lib/systemd/system/ssh.service
[Unit]
Description=OpenBSD Secure Shell server
Documentation=man:sshd(8) man:sshd_config(5)
After=network.target auditd.service
ConditionPathExists=!/etc/ssh/sshd_not_to_be_run

[Service]
EnvironmentFile=-/etc/default/ssh
ExecStartPre=/usr/sbin/sshd -t
ExecStart=/usr/sbin/sshd -D $SSHD_OPTS
ExecReload=/usr/sbin/sshd -t
ExecReload=/bin/kill -HUP $MAINPID
KillMode=process
Restart=on-failure
RestartPreventExitStatus=255
Type=notify
RuntimeDirectory=sshd
RuntimeDirectoryMode=0755

[Install]
WantedBy=multi-user.target
Alias=sshd.service
$ systemctl cat ssh.socket
# /lib/systemd/system/ssh.socket
[Unit]
Description=OpenBSD Secure Shell server socket
Before=ssh.service
Conflicts=ssh.service
ConditionPathExists=!/etc/ssh/sshd_not_to_be_run

[Socket]
ListenStream=22
Accept=yes

[Install]
WantedBy=sockets.target
```

Reply:
I get the following error: `Failed to enable unit, refusing to operate on linked unit file sshd.service`

Reply:
Yeah, I get the same. I guess you can't enable/disable aliases, only actual units.